src/hotspot/share/memory/arena.hpp

```diff
@@ -32,10 +32,9 @@
 
 #include <new>
 
-// The byte alignment to be used by Arena::Amalloc.  See bugid 4169348.
-// Note: this value must be a power of 2
-
-#define ARENA_AMALLOC_ALIGNMENT (2*BytesPerWord)
+// The byte alignment to be used by Arena::Amalloc.
+#define ARENA_AMALLOC_ALIGNMENT BytesPerLong
+static_assert(is_power_of_2(ARENA_AMALLOC_ALIGNMENT), "is aligned");
```

Comment:
This static_assert could just be dropped, along with ARENA_ALIGN_M1 and ARENA_ALIGN_MASK, and change ARENA_ALIGN to
`#define ARENA_ALIGN(x) (align_up((x), ARENA_AMALLOC_ALIGNMENT))`
or replace the macro with
`constexpr size_t arena_align(size_t x) { return align_up(x, ARENA_AMALLOC_ALIGNMENT); }`
(and either way, #include align.hpp).

Reply:
sure, makes sense.